### DISCUSSION TITLE
Why are `memmap` files saved twice during motion correction?

### DISCUSSION THREAD

**Discussion by programerkotik** [flatironinstitute/CaImAn#1425] [Q&A]
When running the parts of codes from the CNMF-E demo notebook for motion correction part, I observed that `memmap` files are saved twice:  
1. First during the call to `motion_correct(save_movie=True)`.  
2. Again explicitly via `cm.save_memmap`.  

Here:
```
%%time
if motion_correct:
    # do motion correction rigid
    mot_correct = MotionCorrect(movie_path, dview=cluster, **parameters.get_group('motion'))
    mot_correct.motion_correct(save_movie=True)
    fname_mc = mot_correct.fname_tot_els if pw_rigid else mot_correct.fname_tot_rig
    if pw_rigid:
        bord_px = np.ceil(np.maximum(np.max(np.abs(mot_correct.x_shifts_els)),
                                     np.max(np.abs(mot_correct.y_shifts_els)))).astype(int)
    else:
        bord_px = np.ceil(np.max(np.abs(mot_correct.shifts_rig))).astype(int)
        # Plot shifts
        plt.plot(mot_correct.shifts_rig)  # % plot rigid shifts
        plt.legend(['x shifts', 'y shifts'])
        plt.xlabel('frames')
        plt.ylabel('pixels')
        plt.gcf().set_size_inches(6,3)

    bord_px = 0 if border_nan == 'copy' else bord_px
    fname_new = cm.save_memmap(fname_mc, base_name='memmap_', order='C',
                               border_to_0=bord_px)
else:  # if no motion correction just memory map the file
    fname_new = cm.save_memmap(movie_path, base_name='memmap_',
                               order='C', border_to_0=0, dview=dview)
```

This redundancy becomes problematic with large datasets and it seems unnecessary to save the data twice, yet my attempts to bypass one of the saves lead to issues.  

### **Question**  
Why are `memmap` files saved twice during motion correction?  

The design seems intentional, but I wonder why. I noticed that one of them is saved in "F" order, and another one in "C". What is the meaning of that?

**Comment by kushalkolar** [comment]
Probably saving the memmap in C-order to run CNMF? There's no saving of memmaps twice for motion correction, [mescore doesn't do it](https://github.com/nel-lab/mesmerize-core/blob/c7d5df6b2c12c935505e7e54d38649897e2d13b3/mesmerize_core/algorithms/mcorr.py#L62-L77), the `save_movie=True` is how it's done.

**Reply by echoer0925** [reply]
I have a strange problem. In demo_online_cnmfE notebooks, after saving the F_frames mmap files one-by-one , but when run the code `fname_new = cm.save_memmap(mc.mmap_file, base_name="memmap_", order="C", border_to_0=0, dview=dview`, the ubuntu system always break down and automatically restart. All the variables are lost so I have to repeat again. Could you tell me how I can solve this situation.

**Reply by pgunn** [reply]
> I have a strange problem. In demo_online_cnmfE notebooks, after saving the F_frames mmap files one-by-one , but when run the code `fname_new = cm.save_memmap(mc.mmap_file, base_name="memmap_", order="C", border_to_0=0, dview=dview`, the ubuntu system always break down and automatically restart. All the variables are lost so I have to repeat again. Could you tell me how I can solve this situation.

There's a good chance it's running out of memory or storage; if you can keep an eye on the memory use with htop or something similar that may help us figure out what's going on.

Is this a VM or is Ubuntu running on raw hardware?

**Reply by echoer0925** [reply]
thank you,  i have solved this problem. my ubuntu computer has a low memory use. Since i add the memory up, the problem is solved.
